datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/exception/DataHubDataFetcherExceptionHandler.java

```diff
@@ -24,23 +24,33 @@ public DataFetcherExceptionHandlerResult onException(
     DataHubGraphQLErrorCode errorCode = DataHubGraphQLErrorCode.SERVER_ERROR;
     String message = "An unknown error occurred.";
 
-    // note: make sure to access the true error message via `getCause()`
-    if (exception.getCause() instanceof IllegalArgumentException) {
+    IllegalArgumentException illException =
+        findFirstThrowableCauseOfClass(exception, IllegalArgumentException.class);
+    if (illException != null) {
+      log.error("Illegal Argument Exception");
```

Comment:
this log line is not useful - please remove or add the context of the underlying exception:

log.error("Received illegal argument exception", illException);